src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type == Bulldozer.class) {
+            return (List<T>) new BulldozerProducer().get();
+        }
+        if (type == Excavator.class) {
+            return (List<T>) new ExcavatorProducer().get();
+        }
+        if (type == Truck.class) {
+            return (List<T>) new TruckProducer().get();
+        }
+        return new ArrayList<>();
```

Comment:
This code must be a good example of a polymorphism.
Let's create an abstract variable of a type MachineProducer and initialize it with an applicable producater implementation. Then just call the method get one time.

------
(optional) Also, you can try to use a hashMap instead of if-blocks. It is a really good practice.

Reply:
> This code must be a good example of polymorphism. Let's create an abstract variable of a type MachineProducer and initialize it with an applicable producer implementation. Then just call the method get one time.
> 
> (optional) Also, you can try to use a hashMap instead of if-blocks. It is a really good practice.

Good idea